Core/Graphics/Source/Graphics.cpp

```diff
@@ -88,56 +174,37 @@ namespace Babylon
         }
         else
         {
-            return whenAllTask.then(Dispatcher, arcana::cancellation::none(), [this, &finished, &workDone]() mutable {
+            return whenAllTask.then(m_renderWorkDispatcher, arcana::cancellation::none(), [this, &finished, &workDone]() mutable {
```

Comment:
This is a nice once! `finished` is created on the stack. But when adding a new task, its reference is captured but the stack changes when this lambda is executed. And &finished points somewhere random on the stack. and we get a dead lock because finished is never really true.

Reply:
I don't think the mechanism quite works the way you describe. `&finished` does point somewhere on the stack, but it's not random. The local variable targeted by that reference is in the same function that's ticking the dispatcher which is driving this asynchronous work. Furthermore, that ticking is being done in a `while` loop that cannot leave until `finished` is set to true, at which point the dispatcher is guaranteed to be empty of all tasks which ever took a reference to that local variable. In sum, it should be impossible for the part of the stack which defines `finished` to change while any tasks refer to it because, in order to run those tasks, we only go _in_ from that scope, not _out_ from it.

If you're still seeing a deadlock, please let me know. I'm not reproing a deadlock on macOS right now, and I doubt the true cause would come from here, but it's definitely something that needs investigating if we're running into it.